### ISSUE TITLE
Lecture fichier BLOB

### ISSUE THREAD

**Issue by jyknecht** [witsa/synapps#418]
Bonjour,
j'ai un fichier blob (dynamique) formaté pour les options d'une liste de choix déroulant, comment le récupérer pour le lier à l'acteur ?
Cordialement

**Comment by WITNicolas** [comment]
Salut,
Peux tu me transmettre le BLOB ?
Cordialement,

**Comment by WITNicolas** [comment]
J'ai pris le BLOB du site, je me doutais de l'installation.
Il suffit de créer un acteur Liste et de liaisonner la propriété OPTION avec ta ressource Texte et de rajouter le :state
a priori ça devrait fonctionner.

**Comment by jyknecht** [comment]
j'ai déjà essayer, ca marche jusqu'à 255 caractères.

**Comment by WITNicolas** [comment]
Oui effectivement... je vais rechercher un peu comment faire cela directement ou bien trouver une astuce qui permettra de réaliser la même chose.

**Comment by jyknecht** [comment]
j'ai aussi essayer avec readAsArrayBuffer  et filereader, mais je n'arrive pas à ouvrir le BLOB en dynamique.
je t'envoi le lien site

**Comment by jyknecht** [comment]
MaJ 1.03 
Magasin de donnée REDY : Méthode loadWosVariable() permet l'option withBlob qui permet de charger les valeurs de variable de type BLOB.
Est-ce que cela pourrait répondre à ma question, si oui comment faire ?

**Comment by manufitoussiwit** [comment]
@jyknecht Effectivement, c'est sensé être cette méthode mais actuellement ce n'est pas donctionnel avec les blob fichier. 

En attendant, il y a une autre méthode par script. 
En voici un exemple qui permet d'aller chercher le contenu d'un fichier blob : 

```js
(async () => {
  try {
    // Pour aller chercher un Fichier BLOB à l'adresse  :WEB.IMG.test~txt

    const url = `${this.dataStores.redy.host.domain}/WSID${this.dataStores.redy.host.session.sid}/WEB/IMG/test.txt`;
    const result = await this.dataStores.redy.ajax.fetch({ url });

    // pour récupérer un JSON
    // const json = await result.json();
    // console.log(json);

    // Pour récupérer du texte :
    const txt = await result.text();
    console.log('resultat: ', txt);

  } catch (error) {
    // une erreur est survenue, il faut surement faire quelque chose.
    console.error(error);
  }
})();
```
